apstools/plans/xpcs_mesh.py

```diff
@@ -0,0 +1,237 @@
+from bluesky import plan_stubs as bps
+from bluesky import utils
+from bluesky import plan_patterns
+from bluesky import preprocessors as bpp
+
+import inspect
+from itertools import zip_longest
+from collections import defaultdict
+import os
+
+try:
+    # cytools is a drop-in replacement for toolz, implemented in Cython
+    from cytools import partition
+except ImportError:
+    from toolz import partition
+
+
+def mesh_list_grid_scan(detectors, *args, number_of_collection_points, snake_axes=False, per_step=None, md=None):
+    """
+    Scan over a mesh; each motor is on an independent trajectory.
```

Comment:
Sounds like a grid scan.  Needs a bit more.  Suggestion:

```
Scan over a multi-dimensional mesh, collecting a total of *n* points.

- Each motor is on an independent trajectory.
- Repeat the sequence of mesh positions until *n* points are collected.
```

Reply:
Is the mesh truly multi-dimensional or just 2-D?  Whatever, we should say.

Reply:
Given, that it is copied from list_grid_scan I believe it is truly multi-dimensional. Not sure how it would look in higher n-dimensions but theoretically it should work.